### ISSUE TITLE
Ingest simulate should coerce `_id` to a string

### ISSUE THREAD

**Issue by clintongormley** [elastic/elasticsearch#23823]
Passing an unquoted `_id` to simulate:

```
POST _ingest/pipeline/_simulate
{
  "pipeline": {
    "processors": [
      {
        "set": {
          "field": "foo",
          "value": "bar"
        }
      }
    ]
  },
  "docs": [
    {
      "_index": "t",
      "_type": "t",
      "_id": 1,
      "_source": {
        "foo": 1
      }
    }
  ]
}
```

throws the following exception:

```
      {
        "type": "parse_exception",
        "reason": "[_id] property isn't a string, but of type [java.lang.Integer]",
        "header": {
          "property_name": "_id"
        }
      }
 ```

The `_id` should be coerced to a string.  In fact, so should `_index`, `_type`, `_routing`, `_parent`...

**Comment by sneivandt** [comment]
Is this what you are expecting for this case? I would also ask if it is frowned upon for me to submit PRs for multiple issues tagged "low hanging fruit" rather than letting other new contributors take them?

**Comment by talevy** [comment]
Closed by: #23885 
thanks @sneivandt!
